lib/kino_vega_lite/chart_cell.ex

```diff
@@ -406,4 +439,27 @@ defmodule KinoVegaLite.ChartCell do
       _ -> nil
     end
   end
+
+  defp infer_types({:columns, %{columns: _columns}, data}) do
+    Enum.map(data, fn data -> Enum.to_list(data) |> List.first() |> type_of() end)
```

Comment:
```suggestion
    Enum.map(data, fn data -> data |> Enum.at(0) |> type_of() end)
```

Otherwise we traverse the whole enumerable :)